docs/usage/ipv6.md

```diff
@@ -0,0 +1,154 @@
+This allows the cluster to resolve IPv4-only DNS names and to connect to IPv4-only services.
+
+Please note that traffic going through NAT64 incurs the same cost as ordinary NAT traffic in an IPv4-only cluster.
+Therefore, it might be beneficial to prefer IPv6 for services, which provide IPv4 and IPv6.
+
+## Dual-Stack Shoot Clusters
+
+### Motivation
+
+Dual-stack shoot clusters support IPv4 and IPv6 out-of-the-box.
+They might also be the obvious intermediate step on the way towards IPv6 for any existing (IPv4-only) clusters.
+
+### Creating a Dual-Stack Shoot Cluster
+
+To create a dual-stack shoot cluster, the following needs to be specified in the `Shoot` resource:
+
+```yaml
+kind: Shoot
+apiVersion: core.gardener.cloud/v1beta1
+metadata:
+  ...
+spec:
+  ...
+  networking:
+    type: ...
+    pods: 192.168.128.0/17
+    nodes: 192.168.0.0/18
+    services: 192.168.64.0/18
+    ipFamilies:
+      - IPv4
+      - IPv6
+  ...
+  provider:
+    type: aws
+    infrastructureConfig:
+      apiVersion: aws.provider.extensions.gardener.cloud/v1alpha1
+      kind: InfrastructureConfig
+      networks:
+        vpc:
+          cidr: 192.168.0.0/18
+        zones:
+          - name: ...
+            workers: 192.168.0.0/19
+            public: 192.168.32.0/20
+            internal: 192.168.48.0/20
+```
+
+Please note that the only change compared to an IPv4-only shoot cluster is the addition of `IPv6` to the `.spec.networking.ipFamilies` field.
+The order of the IP families defines the preference of the IP family.
+In this case, IPv4 is preferred over IPv6, e.g. services specifying no IP family will get only an IPv4 address.
+
+### Migration of IPv4-only Shoot Clusters to Dual-Stack
+
+Eventually, migration should be as easy as changing the `.spec.networking.ipFamilies` field in the `Shoot` resource from `IPv4` to `IPv4, IPv6`.
+However, as of now, this is not supported.
+
+It is worth recognizing that the migration from an IPv4-only shoot cluster to a dual-stack shoot cluster involves rolling of the nodes/workload as well.
+Nodes will not get a new IPv6 address assigned automatically.
+The same is true for pods as well.
+Once the migration is supported, the detailed caveats will be documented here.
+
+### Load Balancer Configuration
+
+The AWS Load Balancer Controller is automatically deployed when using a dual-stack shoot cluster.
+When creating a load balancer, the corresponding annotations need to be configured, see [AWS Load Balancer Documentation - Network Load Balancer](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/service/nlb/) for details.
+
+The AWS Load Balancer Controller allows dual-stack ingress so that a dual-stack shoot cluster can serve IPv4 and IPv6 clients.
+You can find an example [here](dual-stack-ingress.md#creating-an-ipv4ipv6-dual-stack-ingress).
+
+Please note that load balancer services without any special annotations will default to IPv4-only regardless how `.spec.ipFamilies` is set.
```

Comment:
Maybe a warning here too and potentially moving it a bit up at the point where you mention that they need to use the NLB annotations.